ihontaryk/homework_8/restaurant.py

```diff
@@ -0,0 +1,78 @@
+from collections import OrderedDict
+from functools import partialmethod
+
+
+class Restaurant:
+    menu = {'first_course': 'First Course',
+            'second_course': 'Second Course',
+            'dessert': 'Dessert',
+            'beverage': 'Beverage'}
+
+    def __init__(self):
+        self.order = OrderedDict()
+
+    def _add_item(self, key, count, price):
+        if key not in self.menu:
+            raise ValueError(f'{key} is not available in the menu')
+
+        name = self.menu[key]
+
+        if name not in self.order:
+            self.order[name] = 0
+        self.order[name] += count * price
+
+    _add_first_course = partialmethod(_add_item, "first_course")
+    _add_second_course = partialmethod(_add_item, "second_course")
+    _add_dessert = partialmethod(_add_item, "dessert")
+    _add_beverage = partialmethod(_add_item, "beverage")
```

Comment:
I do not see much value in these methods, which do not help with scaling
_add_item is enough, 
maybe you could introduce class to store dish types

```
class DishType:
    BEVARAGE = "bevarage"
    FIRST_COURSE = "first_course"



add_item(DishType.BEVARAGE, 2, 10)



from collection import namedtuple

OrderItem = namedtuple("OrderItem", ["kind", "count", "price"])
item = OrderItem(DishType.BEVARAGE, 2, 10)
item.kind
item.count
item.price

add_item(*item)

```